src/meta/src/barrier/mod.rs

```diff
@@ -160,6 +161,10 @@ pub struct GlobalBarrierManager<S: MetaStore> {
     metrics: Arc<MetaMetrics>,
 
     env: MetaSrvEnv<S>,
+
+    collect_word_node_map: Arc<RwLock<HashMap<WorkerId, Option<CollectOverRequest>>>>,
+
+    collect_over_sender: Arc<RwLock<Option<Sender<Vec<CollectOverRequest>>>>>,
```

Comment:
To ensure linearizability, the global barrier manager was designed in a state-machine style. So I believe there should be as fewer shared memories or locks as possible to make this whole thing more readable and reduce potential race cases.

A more straightforward way to implement this might be starting two tasks that are responsible for injecting and collecting separately, and we can control the number of concurrent checkpoints simply by something like sharing a semaphore between them.

Reply:
Yes，I think my code is not good. I will separate inject and collect in loacl_stram_manager like Patrick said. inject_barrier will send to actor and save it rx in map. then collect_barrier will find rx and rx.await.